frontend/src/components/pages/RegistrantExperience/PersonalInfo/CamperCard.tsx

```diff
@@ -0,0 +1,327 @@
+import React, { useEffect, useState } from "react";
+import { DeleteIcon } from "@chakra-ui/icons";
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  Box,
+  Button,
+  Divider,
+  Flex,
+  FormControl,
+  FormLabel,
+  Heading,
+  Input,
+  NumberInputField,
+  NumberInput,
+  Spacer,
+  Text,
+  Textarea,
+  useDisclosure,
+  useToast,
+  Wrap,
+  WrapItem,
+  FormErrorMessage,
+} from "@chakra-ui/react";
+import {
+  PersonalInfoActions,
+  PersonalInfoReducerDispatch,
+} from "../../../../types/PersonalInfoTypes";
+import RequiredAsterisk from "../../../common/RequiredAsterisk";
+import { RegistrantExperienceCamper } from "../../../../types/CamperTypes";
+import { checkAge, checkFirstName, checkLastName } from "./personalInfoReducer";
+
+type CamperCardProps = {
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
+  camper: RegistrantExperienceCamper;
+  camperId: number;
+  setPersonalInfo: (action: PersonalInfoReducerDispatch) => void;
+};
+
+const CamperCard = ({
+  nextBtnRef,
+  camper,
+  camperId,
+  setPersonalInfo,
+}: CamperCardProps): React.ReactElement => {
+  const [isFirstNameInvalid, setIsFirstNameInvalid] = useState<boolean>(false);
+  const [isLastNameInvalid, setIsLastNameInvalid] = useState<boolean>(false);
+  const [isAgeInvalid, setIsAgeInvalid] = useState<boolean>(false);
+
+  useEffect(() => {
+    const updateFormErrorMsgs = () => {
+      console.log(
+        checkFirstName(camper.firstName),
+        checkLastName(camper.lastName),
+        checkAge(camper.age),
+        camper.firstName,
+        !!camper.firstName,
+      );
+      if (!checkFirstName(camper.firstName)) setIsFirstNameInvalid(true);
+      if (!checkLastName(camper.lastName)) setIsLastNameInvalid(true);
+      if (!checkAge(camper.age)) setIsAgeInvalid(true);
+    };
+
+    if (nextBtnRef && nextBtnRef.current) {
+      // Passing the same reference
+      nextBtnRef.current.addEventListener("click", updateFormErrorMsgs);
+    }
+
+    return () => {
+      // Passing the same reference
+      if (nextBtnRef && nextBtnRef.current) {
+        nextBtnRef.current.removeEventListener("click", updateFormErrorMsgs);
+      }
+    };
+  }, [camper]);
+
+  function AlertDialogExample() {
```

Comment:
Could we rename this to `DeleteRegistrantConfirmationModal` and shift it to it's own file? I personally prefer modals over AlertDialogues since that seems to be the standard we're using across the project. Happy to change my mind if there was a particular reason you used AlertDialogues instead of modals.